python/pyspark/statcounter.py

```diff
@@ -114,24 +120,24 @@ def variance(self):
     # Return the sample variance, which corrects for bias in estimating the variance by dividing
     # by N-1 instead of N.
     #
-    def sampleVariance(self):
+    def sampleVariance(self) -> float:
         if self.n <= 1:
             return float('nan')
         else:
             return self.m2 / (self.n - 1)
 
     # Return the standard deviation of the values.
-    def stdev(self):
+    def stdev(self) -> float:
         return sqrt(self.variance())
 
     #
     # Return the sample standard deviation of the values, which corrects for bias in estimating the
     # variance by dividing by N-1 instead of N.
     #
-    def sampleStdev(self):
+    def sampleStdev(self) -> float:
         return sqrt(self.sampleVariance())
 
-    def asDict(self, sample=False):
+    def asDict(self, sample: bool = False) -> Dict[str, Union[float, int]]:
```

Comment:
Nit. Might be cleaner to have `Dict[str, float]`.

Reply:
@zero323 I think we should keep ```Dict[str, Union[float, int]]```  because of ```'count': self.count()``` is ```int``` in the return. WDYT?

Reply:
There are contexts, where distinguishing between these two is important (for us, it is primarily for arguments which are passed to JVM), however every integer is a real number (see [Numerical tower](https://en.wikipedia.org/wiki/Numerical_tower) and, albeit not that useful in practice,  [PEP 3141](https://www.python.org/dev/peps/pep-3141/)).

Reply:
updated. Thanks!